lib/javascripts/base_app.js

```diff
@@ -68,6 +68,10 @@ function BaseApp(zafClient, data) {
   let evt = { firstLoad: true };
   resolveHandler(this, 'app.created')();
   resolveHandler(this, 'app.activated')(evt, evt);
+  $(window).unload(() => {
+    resolveHandler(this, 'app.willDestroy')();
+    delete this.events['app.willDestroy']; // make sure this event only fires once
```

Comment:
Not sure it does that either, since the event has already been bound at this point 😸